packages/authentication/src/services/user-identity.service.ts

```diff
@@ -0,0 +1,61 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: @loopback/authentication
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+/**
+ * The User Identity service links a user to profiles from an external source (eg: ldap, oauth2 provider, saml)
+ * which can identify the user. The profile typically has the following information:
+ *   name, email-id, uuid, roles, authorizations, scope of accessible resources, expiration time for given access
+ *
+ * @example
+ *  export class LDAPUserIdentityService implements UserIdentityService<LDAPUserIdentity, UserProfile> {
+ *    constructor(
+ *      @repository(UserRepository)
+ *      public userRepository: UserRepository,
+ *      @repository(UserIdentityRepository)
+ *      public userIdentityRepository: UserIdentityRepository,
+ *    ) {}
+ *  }
+ */
+export interface UserIdentityService<I, U> {
+  /**
+   * find or create a local user using a profile from an external source
+   * @param userIdentity
```

Comment:
maybe write it more clear that `userIdentity` here refers to the external identity.